plugins/processors/t128_transform/t128_transform.go

```diff
@@ -77,36 +81,35 @@ func (r *T128Transform) Apply(in ...telegraf.Metric) []telegraf.Metric {
 				continue
 			}
 
-			currentValue, converted := convert(field.Value)
-			if !converted {
-				r.Log.Warnf("Failed to convert field '%s' to float for transformation. This transformation will be skipped.")
-				continue
-			}
-
 			cacheFields, metricIsCached := r.cache[seriesHash]
 			if !metricIsCached {
 				r.cache[seriesHash] = make(map[string]observedValue, 0)
 			}
 
-			itemAdded := false
-			if observed, ok := cacheFields[field.Key]; ok {
-				if point.Time().Before(observed.expires) {
-					value, err := r.transform(observed.timestamp, point.Time(), observed.value, currentValue)
-					if err != nil {
-						r.Log.Warnf("excluding failed transform: %v", err)
-					} else {
-						point.AddField(target.key, value)
-						itemAdded = true
-					}
+			observed, ok := cacheFields[field.Key]
+			if !ok {
+				observed = observedValue{
+					value: nil,
 				}
 			}
 
-			if (target.matchesSource && !itemAdded) || (!target.matchesSource && r.RemoveOriginal) {
+			expired := !point.Time().Before(observed.expires)
+
+			itemTransformed := false
+			value, err := r.transform(expired, observed.timestamp, point.Time(), observed.value, field.Value)
+			if err != nil {
+				r.Log.Warnf("excluding failed transform: %v", err)
+			} else if value != nil {
+				itemTransformed = true
+				point.AddField(target.key, value)
+			}
+
+			if (target.matchesSource && !itemTransformed) || (!target.matchesSource && r.RemoveOriginal) {
```

Comment:
There is a lot going on in this func so it a little hard to follow. Maybe some intermediate vars or helper funcs to help clarify the different parts? No biggie

Reply:
I honestly tried to clean it up and there are enough moving pieces that it becomes more complicated. I'm going to leave it the way it is for now.

Reply:
That is fine. Thanks for trying